config/metrics/crd-metrics-config.yaml

```
@@ -0,0 +1,676 @@
+---
+kind: CustomResourceStateMetrics
+spec:
+  resources:
+  - errorLogV: 0
+    groupVersionKind:
+      group: infrastructure.cluster.x-k8s.io
+      kind: VSphereCluster
+      version: v1beta1
+    labelsFromPath:
+      cluster_name:
+      - metadata
+      - labels
+      - cluster.x-k8s.io/cluster-name
+      name:
+      - metadata
+      - name
+      namespace:
+      - metadata
+      - namespace
+      uid:
+      - metadata
+      - uid
+    metricNamePrefix: capv_vspherecluster
```

```suggestion
    metricNamePrefix: capi_vspherecluster
```
I wonder if we should use capi as a prefix.

I'm wondering if we can make the core CAPI dashboard work across infra providers in a generic way. I think technically it should be possible, but it would help if the differences between metrics are as minimal as possible

Let's leave it as is for now. I would experiment a bit with the Grafana dashboard once we get closer to merging this PR (after merging the gen tool I guess)

---

Resolving here in favor of the other PR